azurerm/resource_arm_dns_cname_record_test.go

```diff
@@ -116,42 +117,49 @@ func testCheckAzureRMDnsCNameRecordExists(name string) resource.TestCheckFunc {
 			return fmt.Errorf("Not found: %s", name)
 		}
 
-		conn := testAccProvider.Meta().(*ArmClient).rivieraClient
-
-		readRequest := conn.NewRequestForURI(rs.Primary.ID)
-		readRequest.Command = &dns.GetCNAMERecordSet{}
+		cnameName := rs.Primary.Attributes["name"]
+		zoneName := rs.Primary.Attributes["zone_name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for DNS CNAME record: %s", cnameName)
+		}
 
-		readResponse, err := readRequest.Execute()
+		conn := testAccProvider.Meta().(*ArmClient).dnsClient
+		resp, err := conn.Get(resourceGroup, zoneName, cnameName, dns.CNAME)
 		if err != nil {
-			return fmt.Errorf("Bad: GetCNAMERecordSet: %s", err)
+			return fmt.Errorf("Bad: Get CNAME RecordSet: %v", err)
 		}
-		if !readResponse.IsSuccessful() {
-			return fmt.Errorf("Bad: GetCNAMERecordSet: %s", readResponse.Error)
+
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: DNS CNAME record %s (resource group: %s) does not exist", cnameName, resourceGroup)
 		}
 
 		return nil
 	}
 }
 
 func testCheckAzureRMDnsCNameRecordDestroy(s *terraform.State) error {
-	conn := testAccProvider.Meta().(*ArmClient).rivieraClient
+	conn := testAccProvider.Meta().(*ArmClient).dnsClient
 
 	for _, rs := range s.RootModule().Resources {
 		if rs.Type != "azurerm_dns_cname_record" {
 			continue
 		}
 
-		readRequest := conn.NewRequestForURI(rs.Primary.ID)
-		readRequest.Command = &dns.GetCNAMERecordSet{}
+		cnameName := rs.Primary.Attributes["name"]
+		zoneName := rs.Primary.Attributes["zone_name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := conn.Get(resourceGroup, zoneName, cnameName, dns.CNAME)
 
-		readResponse, err := readRequest.Execute()
 		if err != nil {
-			return fmt.Errorf("Bad: GetCNAMERecordSet: %s", err)
+			return nil
```

Comment:
I think we should be checking for the 404 above this, and then returning the error if it's present - rather than hiding this?